src/main/java/ch/njol/skript/effects/EffDetonate.java

```diff
@@ -0,0 +1,69 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.Firework;
+import org.bukkit.entity.Projectile;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Detonate Fireworks")
+@Description("Forces fireworks to explode immediately, as if they have no remaining fuses.")
```

Comment:
A note should be added to state that when detonating a firework in 1.13 (maybe 1.14 not sure) that if the firework was not atleast level 1 in duration, that it will not detonate but rather remove itself from the world without an explosion.

Reply:
Placing this PR in draft due to some issue with a workaround of this for 1.13 support (Expression#getAll(Event) returns an empty array somehow) or we simply exclude this feature to 1.14 and above.